src/ess/imaging/io.py

```diff
@@ -0,0 +1,265 @@
+# SPDX-License-Identifier: BSD-3-Clause
+# Copyright (c) 2024 Scipp contributors (https://github.com/scipp)
+
+from collections.abc import Callable, Iterable
+from enum import Enum
+from itertools import pairwise
+from pathlib import Path
+from typing import NewType
+
+import jax.numpy as np
+import scipp as sc
+from h5py import File
+from tifffile import imwrite
+
+from ess.reduce.nexus.types import FilePath
+
+from .types import (
+    DEFAULT_HISTOGRAM_PATH,
+    HistogramModeDetectorsPath,
+    ImageDetectorName,
+    ImageKeyLogs,
+    RotationLogs,
+    RotationMotionSensorName,
+)
+
+HistogramModeDetectorData = NewType("HistogramModeDetectorData", sc.DataArray)
+"""Histogram mode detector data."""
+ImageKeyCoord = NewType("ImageKeyCoord", sc.Variable)
+"""Image key coordinate."""
+ImageStacks = NewType("ImageStacks", sc.DataArray)
+"""Image stacks separated by ImageKey values via timestamp."""
+RotationAngleCoord = NewType("RotationAngleCoord", sc.Variable)
+"""Rotation angle coordinate."""
+
+
+class ImageKey(Enum):
+    """Image key values."""
+
+    SAMPLE = 0
+    DARK_CURRENT = 1
+    OPEN_BEAM = 2
+
+
+def load_nexus_histogram_mode_detector_data(
+    *,
+    file_path: FilePath,
+    image_detector_name: ImageDetectorName,
+    histogram_mode_detectors_path: HistogramModeDetectorsPath = DEFAULT_HISTOGRAM_PATH,
+) -> HistogramModeDetectorData:
+    with File(file_path, mode="r") as f:
```

Comment:
Please use `h5py.File` for clarity which `File` this is.

If this is NeXus, why can't we use ScippNexus?